x-pack/plugins/endpoint/public/applications/endpoint/models/index_pattern.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { all } from 'deepmerge';
+import { Immutable } from '../../../../common/types';
+import { IIndexPattern } from '../../../../../../../src/plugins/data/common';
+
+/**
+ * Model for the `IIndexPattern` interface exported by the `data` plugin.
+ */
+export function clone(value: IIndexPattern | Immutable<IIndexPattern>): IIndexPattern {
+  return all([value]) as IIndexPattern;
```

Comment:
@paul-tavares i'm on the same page as you re: using libraries to clone and/or support other immutable patterns. I'm using `all` from `deepmerge` here. The one in lodash had too much code for me to read and understand